packages/core/src/server-runtime-client.ts

```diff
@@ -257,7 +261,7 @@ export class ServerRuntimeClient<
 
     const span = scope.getSpan();
     if (span) {
-      const samplingContext = span.transaction ? span.transaction.getDynamicSamplingContext() : undefined;
+      const samplingContext = span.transaction ? getDynamicSamplingContextFromSpan(span) : undefined;
```

Comment:
```suggestion
      const samplingContext = span.transaction ? getDynamicSamplingContextFromSpan(span.transaction) : undefined;
```

I guess that still needs to come from the transaction?

Reply:
Hmm I see now, this looks at the span transaction under the hood 🤔 I guess that's OK - TBH not sure if this should be in the utility function or not 🤔

Reply:
@myda good point, thanks for raising this. However, I'd say we should be able to pass any span into this function and internally, we obtain the root span (or txn for now). I changed this in [75134c4](https://github.com/getsentry/sentry-javascript/pull/10094/commits/75134c42439872dca8d6a21163190bbcb6543301) - WDYT?

Reply:
I think this is OK - and an implemention detail then I think, so we may change this if needed transparently! 👍